wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -375,3 +396,11 @@ private class SourceCodeHelper(
   }
 }
 
+// For helping with parsing map fields.
+class MapEntryType(
```

Comment:
Could also just store the string representation of the map type as the value in `mapTypes` above and not have to deal with creating these intermediate tuple values.